physical/raft/fsm.go

```diff
@@ -68,6 +78,22 @@ func (f *FSMEntry) String() string {
 	return fmt.Sprintf("Key: %s. Value: %s", f.Key, hex.EncodeToString(f.Value))
 }
 
+func (f *FSMEntry) IsTxError() bool {
+	return f.Key == fsmEntryTxErrorKey
+}
+
+func (f *FSMEntry) AsTxError() error {
+	str := string(f.Value)
+	commitErr := physical.ErrTransactionCommitFailure.Error()
+
+	split := strings.SplitN(str, commitErr, 1)
+	if len(split) != 2 {
```

Comment:
@cipherboy : I'm not sure about this: we should rather have strings.SplitN(str, commitErr, **2**) ?

Reply:
Ah, good catch, playground confirms you are right: https://go.dev/play/p/d3_hn453lQN

Fortunately this just means `errors.Is(...)` isn't true and so is relatively minor, but I will address it in a follow-up PR.

Reply:
https://github.com/openbao/openbao/pull/466